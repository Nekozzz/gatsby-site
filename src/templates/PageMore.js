import React from "react"
import {graphql, Link} from "gatsby"
import SiteLayout from "../components/SiteLayout.js"
import ItemFull from "../components/Item/ItemFull";
import SEO from "../components/seo"

export const query = graphql`
  query($car_id: Int!) {
    hasura {
      car_model(where: {id: {_eq: $car_id}}) {
        id
        slug
        model_name
        model_description
        id
        car_brand {
          brand_name
          slug
        }
      }
    }
  }
`;

const PageMore = ({ data }) => {
  const car = data.hasura.car_model[0];

  return (
    <SiteLayout selectedMenuItem={car.car_brand.slug}>
      <SEO title={car.model_name} />

      <div style={{
        'marginBottom': '8px'
      }}>
        <Link to={`/${car.car_brand.slug}`}>Каталог {car.car_brand.brand_name}</Link>
      </div>

      <hr  style={{
        'marginBottom': '14px'
      }}/>

      <ItemFull car={car} />

    </SiteLayout>
  )
};

export default PageMore