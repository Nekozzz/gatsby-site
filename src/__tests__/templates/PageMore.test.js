import React from "react"
import renderer from "react-test-renderer"
import PageMore from "../../templates/PageMore"
import * as Gatsby from 'gatsby';

describe("PageMore", () => {
  test("renders correctly", () => {

    jest.spyOn(Gatsby, 'useStaticQuery').mockImplementation(() => ({
      site: {
        siteMetadata:{
          title: 'asd'
        }
      },
      hasura: {
        car_brands: [
          {
            slug: 'adidas',
            brand_name: 'Adiads'
          },
          {
            slug: 'cat',
            brand_name: 'Cat man'
          },
        ]
      }
    }));

    const graphqlData = {
      hasura: {
        "car_model": [
          {
            "id": 3,
            "slug": "bmw-x5-ii-30i",
            "model_name": "BMW X5 II 30i",
            "car_brand": {
              "brand_name": "BMW",
              "slug": "bmw"
            }
          },
        ]
      }
    }

    const tree = renderer
      .create(
        <PageMore
          data={graphqlData}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
