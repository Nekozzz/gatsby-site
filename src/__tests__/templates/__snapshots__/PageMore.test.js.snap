// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMore renders correctly 1`] = `
Array [
  <header
    style={
      Object {
        "background": "rebeccapurple",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          asd
        </a>
      </h1>
    </div>
  </header>,
  <main
    className="main-container"
  >
    <section
      className="ant-layout ant-layout-has-sider"
    >
      <aside
        className="ant-layout-sider ant-layout-sider-light"
        style={
          Object {
            "flex": "0 0 200px",
            "maxWidth": "200px",
            "minWidth": "200px",
            "width": "200px",
          }
        }
      >
        <div
          className="ant-layout-sider-children"
        >
          <ul
            className="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
            onMouseEnter={[Function]}
            onTransitionEnd={[Function]}
            role="menu"
            style={Object {}}
          >
            <li
              className="ant-menu-item ant-menu-item-only-child"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="menuitem"
              style={
                Object {
                  "paddingLeft": 24,
                }
              }
            >
              <a
                href="/"
              >
                Главная страница
              </a>
            </li>
            <li
              className="ant-menu-item ant-menu-item-only-child"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="menuitem"
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "paddingLeft": 24,
                }
              }
            >
              <a
                href="/adidas"
              >
                Adiads
              </a>
            </li>
            <li
              className="ant-menu-item ant-menu-item-only-child"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="menuitem"
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "paddingLeft": 24,
                }
              }
            >
              <a
                href="/cat"
              >
                Cat man
              </a>
            </li>
          </ul>
        </div>
      </aside>
      <section
        className="ant-layout site-layout"
      >
        <main
          className="ant-layout-content"
          style={
            Object {
              "margin": "24px 16px",
              "minHeight": 280,
              "padding": 24,
            }
          }
        >
          <div>
            <div
              style={
                Object {
                  "marginBottom": "8px",
                }
              }
            >
              <a
                href="/bmw"
              >
                Каталог 
                BMW
              </a>
            </div>
            <hr
              style={
                Object {
                  "marginBottom": "14px",
                }
              }
            />
            <div>
              <img
                alt=""
                className="img-preview"
                src="http://lorempixel.com/460/230/transport/3/"
              />
            </div>
            <div>
              Марка: 
              BMW
            </div>
            <div>
              Модель: 
              BMW X5 II 30i
            </div>
            <div
              style={
                Object {
                  "marginTop": "8px",
                }
              }
            >
              Описание:
              <div />
            </div>
          </div>
        </main>
      </section>
    </section>
  </main>,
  <footer>
    © 
    2020
    , Built with
     
    <a
      href="https://www.gatsbyjs.org"
    >
      Gatsby
    </a>
  </footer>,
]
`;
